### PR TITLE
docs: add apache reverse-proxy example

### LINKED ISSUE #6086: docs: Apache/NGINX reverse proxy configs
We get a lot of questions in Discord about this!

### PR DESCRIPTION
**Pull request by matifali**
This PR adds an Apache reverse-proxy example to deploy Coder with a wildcard tls certificate using LetsEncrypt Certbot.

This will close https://github.com/coder/coder/issues/6086.

cc: @bpmct

Tested with a wildcard subdomain.

### CODE REVIEW

File: examples/web-server/apache/README.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+1. Create Apache configuration for Coder:
+
+   ```console
+   sudo nano /etc/apache2/sites-available/coder.conf
+   ```
+
+2. Add the following content:
+
+   ```apache
+    <VirtualHost *:443>
```
Comment (bpmct) [inline examples/web-server/apache/README.md:76]: I noticed this doesn't include the HTTP redirect, so when I visit http://coder-apache.bpmct.net, I get "Apache2 Default Page" even with the 000-default site disabled.

Can you change this to add a HTTP->HTTPS redirect, like we do for Caddy (and the upcoming NGINX)

File: examples/web-server/apache/README.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        ProxyRequests Off
+        ProxyPreserveHost On
+
+        RewriteEngine On
+        RewriteCond %{HTTP:Connection} Upgrade [NC]
+        RewriteCond %{HTTP:Upgrade} websocket [NC]
+        RewriteRule /(.*) ws://127.0.0.1:3000/$1 [P,L]
+
+        SSLCertificateFile /etc/letsencrypt/live/dev.dietstyler.com/fullchain.pem
+        SSLCertificateKeyFile /etc/letsencrypt/live/dev.dietstyler.com/privkey.pem
```
Comment (bpmct) [inline examples/web-server/apache/README.md:93]:
```suggestion
        SSLCertificateFile /etc/letsencrypt/live/coder.example.com/fullchain.pem
        SSLCertificateKeyFile /etc/letsencrypt/live/coder.example.com/privkey.pem
```

File: examples/web-server/apache/README.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+   ```console
+   sudo nano /etc/apache2/sites-available/coder.conf
+   ```
+
+2. Add the following content:
+
+   ```apache
+    <VirtualHost *:443>
+        ServerName dev.dietstyler.com
+        ServerAlias *.dev.dietstyler.com
```
Comment (bpmct) [inline examples/web-server/apache/README.md:78]:
```suggestion
        ServerName coder.example.com
        ServerAlias *.coder.example.com
```

**Review by bpmct — CHANGES_REQUESTED** [review]
Tested and it worked! Just requesting some changes to be consistent with our Caddy and NGINX ones :)

Screenshot 2023-02-15 at 7 07 01 AM

File: examples/web-server/apache/README.md (hunk 1 above)
Comment (matifali) [reply examples/web-server/apache/README.md:97]: Please test HTTP to HTTPS. It should work now.

File: examples/web-server/apache/README.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+        ProxyPass / http://127.0.0.1:3000/
+        ProxyPassReverse / http://127.0.0.1:3000/
+        ProxyRequests Off
+        ProxyPreserveHost On
+
+        RewriteEngine On
+        RewriteCond %{HTTP:Connection} Upgrade [NC]
+        RewriteCond %{HTTP:Upgrade} websocket [NC]
+        RewriteRule /(.*) ws://127.0.0.1:3000/$1 [P,L]
```
Comment (bpmct) [inline examples/web-server/apache/README.md:111]:
```suggestion
        # Websockets are required for workspace connectivity
        RewriteCond %{HTTP:Connection} Upgrade [NC]
        RewriteCond %{HTTP:Upgrade} websocket [NC]
        RewriteRule /(.*) ws://127.0.0.1:3000/$1 [P,L]
```

Comment (bpmct) [reply examples/web-server/apache/README.md:111]: Let's do this in the NGINX one as well :)

**Review by bpmct — APPROVED** [review]

File: examples/web-server/apache/README.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+   sudo systemctl stop apache2
+   ```
+
+## Install and configure LetsEncrypt Certbot
+
+1. Install LetsEncrypt Certbot: Refer to the [CertBot documentation](https://certbot.eff.org/instructions?ws=apache&os=ubuntufocal&tab=wildcard). Be sure to pick the wildcard tab and select your DNS provider for instructions to install the necessary DNS plugin.
+
+## Create DNS provider credentials
+
+1. Create an API token for the DNS provider you're using: e.g [CloudFlare](https://dash.cloudflare.com/profile/api-tokens) with the following permissions:
```
Comment (bpmct) [inline examples/web-server/apache/README.md:47]: Can we make this more generic and link to the certbot docs for [DNS providers](https://eff-certbot.readthedocs.io/en/stable/using.html#dns-plugins)?

We can still provide concrete instructions for CloudFlare, just don't want users to be lost!

Comment (matifali) [reply examples/web-server/apache/README.md:47]: sure. I will do that

**Review by matifali — COMMENTED** [review]
LGTM

**Pull request merged**